scripts/flytekit_sagemaker_runner.py

```diff
@@ -6,53 +6,32 @@
 from os import environ
 
 FLYTE_ARG_PREFIX = "--__FLYTE"
-FLYTE_ENV_VAR_PREFIX = "{}_ENV_VAR_".format(FLYTE_ARG_PREFIX)
-FLYTE_CMD_ARG_PREFIX = "{}_CMD_ARG_".format(FLYTE_ARG_PREFIX)
+FLYTE_ENV_VAR_PREFIX = f"{FLYTE_ARG_PREFIX}_ENV_VAR_"
+FLYTE_CMD_PREFIX = f"{FLYTE_ARG_PREFIX}_CMD_"
 FLYTE_ARG_SUFFIX = "__"
 
 parser = argparse.ArgumentParser(description="Running sagemaker task")
-parser.add_argument("--__FLYTE_SAGEMAKER_CMD__", dest="flyte_sagemaker_cmd", help="The entrypoint selector argument")
```

Comment:
This shouldn't be removed. See my previous comment

Reply:
We don't want to assume that the hyperparameters are passed in the order we injected them. What we are doing here is we make sure we know what the command is so that we can reconstruct them in a reasonable way.

Reply:
Admittedly, if we do this, instead of making assumption of SM's behavior, we are making assumption about the format of the command carried by task template's container (http://github.com/lyft/flyteplugins/blob/ad68fea4fc2f19f447567d315b700ea308c12d27/go/tasks/plugins/k8s/sagemaker/utils.go#L287-L300). But I think it's either one way or the other. Making assumption about our own code base seems to be safer to deal with. WDYT @EngHabu ?

Reply:
Let me show you an example of what I mean: 
In our SM flyteplugin, I injected the task template's container `args` first, in the same order as they appear in the task template `args` list. And then I injected env variables.

But looking at the following actual command that SM used to invoke the runner script (though I removed the lyft-internal SHAs and s3 paths), you can probably notice that SM sorts all the arguments based on the alphabetical order of the arguments' names.

```
/usr/bin/python3 /srv/service/venv/bin/flytekit_sagemaker_runner.py --__FLYTE_CMD_ARG_inputs__ s3://<path>/inputs.pb --__FLYTE_CMD_ARG_output_prefix__ s3://<path> --__FLYTE_CMD_ARG_task_module__ workflows.custom_sagemaker_training --__FLYTE_CMD_ARG_task_name__ custom_training_task --__FLYTE_ENV_VAR_FLYTE_INTERNAL_CONFIGURATION_PATH__ /code/flyteplayground/flyte_configs/staging.config --__FLYTE_ENV_VAR_FLYTE_INTERNAL_DOMAIN__ development --__FLYTE_ENV_VAR_FLYTE_INTERNAL_IMAGE__ <image> --__FLYTE_ENV_VAR_FLYTE_INTERNAL_PROJECT__ <project> --__FLYTE_ENV_VAR_FLYTE_INTERNAL_VERSION__ <version> --__FLYTE_SAGEMAKER_CMD__ service_venv+pyflyte-execute
```